docker/staging_test/retrieve_bugs_from_db.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fetch Bugs from from datastore"""
+
+import logging
+import os
+import random
+import json
+
+import osv
+import osv.logs
+
+from google.cloud import ndb
+
+GCP_PROJECT = 'oss-vdb-test'
+BUG_DIR = '/staging_testing/all_bugs'
+
+
+def format_bug_for_output(bug: osv.Bug) -> dict[str, any]:
+  """Outputs ndb bug query results to JSON file
+
+  Args:
+    bug: an `osv.Bug` queried from ndb.
+
+  Returns:
+    A dict storing all the important `Bug` fields that we want to use later
+  """
+
+  affected_fuzzy = None
+  # Store one version for use as the query version later.
+  if len(bug.affected_fuzzy) > 0:
+    version_index = random.randrange(len(bug.affected_fuzzy))
+    affected_fuzzy = bug.affected_fuzzy[version_index]
+
+  return {
+      'db_id': bug.db_id,
+      'purl': bug.purl,
+      'project': bug.project,
+      'ecosystem': bug.ecosystem,
+      'affected_fuzzy': affected_fuzzy
+  }
+
+
+def get_bugs_from_datastore() -> None:
```

Comment:
We should probably cache this in a GCS bucket, otherwise every time this gets run it has to run get bugs from datastore from scratch.

Reply:
It could probably be downloaded directly from the `all.zip` files in the export bucket

Reply:
I will keep the datastore query until we implement a [unified all.zip file ](https://github.com/google/osv.dev/issues/2423)(I plan to do this soon).